climakitae/threshold_tools.py

```diff
@@ -14,41 +14,162 @@
 import hvplot.pandas
 import hvplot.xarray
 import panel as pn
+import statsmodels as sm
 from .utils import _read_ae_colormap
+from warnings import warn
 
 
-def get_ams(da, extremes_type="max"):
-    """Function that converts data into annual maximums
+def calculate_ess(data, nlags=None):
+    """
+    Function for calculating the effective sample size (ESS) of the provided data.
+
+    Input array is assumed to be timeseries data with potential autocorrelation.
```

Comment:
I'd recommend placing a check in here to ensure that it is timeseries data. There's an example in the data_loaders code iirc